src/main/resources/META-INF/resources/primefaces-extensions/codescanner/codescanner-widget.js

```diff
@@ -32,35 +32,38 @@ PrimeFaces.widget.ExtCodeScanner = PrimeFaces.widget.BaseWidget.extend({
 
     start: function() {
         var $this = this;
-        function handleResult(result) {
-            if ($this.hasBehavior('codeScanned')) {
-                var ext = {
-                    params: [{
-                            name: $this.id + '_value',
-                            value: result.text
-                        }, {
-                            name: $this.id + '_format',
-                            value: result.format
-                        }]
-                };
-                $this.callBehavior('codeScanned', ext);
+        function handleDevices(devices) {
+            var deviceId = devices[0].deviceId;
+            $this.codeReader.decodeFromVideoDevice(deviceId, $this.video, handleResult);
+        }
+        function handleResult(result, err) {
+            if (result) {
+                if ($this.hasBehavior('codeScanned')) {
+                    var ext = {
+                        params: [{
+                                name: $this.id + '_value',
+                                value: result.text
+                            }, {
+                                name: $this.id + '_format',
+                                value: result.format
+                            }]
+                    };
+                    $this.callBehavior('codeScanned', ext);
+                }
+                if ($this.cfg.onsuccess) {
+                    eval($this.cfg.onsuccess);
```

Comment:
Don't use `eval` it violates CSP.  I think you want to do a `call` right?

Reply:
Thanks for the pointer.. I was not sure how to do this. I think it's okay right now.

Reply:
Yep that will do it!